mlx/backend/gpu/available.h

```diff
@@ -0,0 +1,9 @@
+// Copyright © 2025 Apple Inc.
+
+#pragma once
+
+namespace mlx::core::gpu {
+
+bool is_available();
```

Comment:
Should it take a device parameter to support multi-device?

Reply:
Good question. I think of this one as more of a general "backend availability" API than a "device" availability. But I haven't really thought through what our API for multi-GPU-device on single node looks like. 

I would say let's cross that bridge when we get there... we can always add an overload of this one / update it since it's a pretty internal API.